src/main/java/com/traverse/storage/neo4j/Neo4jService.java

```diff
@@ -0,0 +1,107 @@
+package com.traverse.storage.neo4j;
+import com.traverse.storage.models.CollectionSearchCriteria;
+import com.traverse.storage.models.FieldSearchCriteria;
+import com.traverse.storage.models.Neo4jNode;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.Result;
+import org.neo4j.driver.Record;
+import org.neo4j.driver.types.Node;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author isfar
+ *
+ * This is the Neo4j Service where standardized search and patch implementations are defined
+ * For any Neo4j collection we have, this service can be used to search through the items or
+ * patch
+ * @param <T> generic type - for example, User or Group
+ */
+@Service
+public class Neo4jService<T> {
+
+    @Autowired
+    private Driver driver;
+
+    /**
+     * This search function allows users to give a formatted criteria and collection. Then query that
+     * neo4j collection based on that criteria.
+     * @param entityClass
+     * @param criteria
+     * @return a list of Neo4jNodes
+     */
+    public List<Neo4jNode> search(Class<T> entityClass, CollectionSearchCriteria criteria) {
+        List<Neo4jNode> results = new ArrayList<>();
+
+        try (Session session = driver.session()) {
+            StringBuilder query;
+            if (criteria.getText() != null) {
+                query = new StringBuilder();
+                query.append(criteria.getFilter() ?
+                        "CALL db.index.fulltext.queryNodes('textIndex', " + "'" + criteria.getText() + "'" + ") YIELD node AS n WHERE " :
```

Comment:
Wheres the repository to handle queries? Therea built in one. Take a look at authService register and login functionality.